### PR TITLE
Passed userId as Props and removed hard coded values from delete modal

### LINKED ISSUE #88: Extend the BoardPage and create Modal View to Delete a Board
## User story

**Who:** **As a** User Owner of a Board
**What:** **I want to**  access the Modal View to Delete a Board
**Why:** **so that I can**  delete my own board. The option accessible from the drop down menu on the Board Page

## Acceptance criteria

The Modal View  should look like the one below:
Screenshot 2021-07-30 at 20 38 24Screenshot 2021-07-30 at 20 37 56

#### -> React Containers and Components 
Create the Modal View using an adapted version of the Modal View - https://github.com/HackYourFuture-CPH/simply-name.it/issues/19
Cross Button - https://github.com/HackYourFuture-CPH/simply-name.it/issues/11
Title
Buttons for Delete and Cancel - https://github.com/HackYourFuture-CPH/simply-name.it/issues/2 (probably needs some adaptation)

#### -> Endpoints
Endpoint to delete a board - https://github.com/HackYourFuture-CPH/simply-name.it/issues/62

### PR DESCRIPTION
**Pull request by SlaviaF**
# Description

Adjustments for #88. Props passed for the userId, and hard coded values are removed from the code .

Fixes # 88

# How to test?

to incorporate error handling from the parent component:
1. const [error, setError] = useState(null);

2. const deleteErrorHandler = (err) => {
    setError(err);
  };
3. Pass the error handler as props with the rest of the props
{modalVisibility && (
            <DeleteBoardModal
              boardInfo={boardInfo}
              modalVisibility={modalVisibility}
              setModalVisibility={setModalVisibility}
              errorPageOnDelete={errorPageOnDelete}
            />
          )}

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have followed the name conventions for CSS Classnames and filenames, Components names and filenames, Style filenames, if you are in doubt check the the project README.MD and here https://github.com/HackYourFuture-CPH/curriculum/blob/master/review/review-checklist.md
- [x] I have commented my code, particularly in hard-to-understand areas, if you code was simple enough mark the box anyway
- [x] I have made corresponding changes to the documentation, if you code was simple enough mark the box anyway
- [x] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

**Review by senner007 — CHANGES_REQUESTED** [review]
Looking good
You also need a deleteHandler here. In here you await the delete function and rethrow the error in a setErrorState function

**Comment by senner007** [comment]
> Looking good
> You also need a deleteHandler here. In here you await the delete function and rethrow the error in a setErrorState function

see #155

**Comment by SlaviaF** [comment]
> > Looking good
> > You also need a deleteHandler here. In here you await the delete function and rethrow the error in a setErrorState function
> 
> see #155


I have called a function(deleteErrorHandler) in the fetch  The function and the state will be passed in the parent component (as mentioned in the PR description). Is it the right way.

File: src/client/containers/DeleteBoardModal/DeleteBoardModal.container.js [hunk 1]
```diff
@@ -5,17 +5,25 @@ import CloseButton from '../../components/CloseButton/CloseButton.component';
...
+
   const deleteBoard = async () => {
     const API_URL = `/api/users/${userId}/boards/${boardId}`;
     try {
-      await fetch(API_URL, { method: 'DELETE' });
+      const response = await fetch(API_URL, { method: 'DELETE' });
       // eslint-disable-next-line no-alert
+      if (!response.ok) {
+        throw new Error(response.status);
+      }
```
Comment (senner007) [inline src/client/containers/DeleteBoardModal/DeleteBoardModal.container.js:21]: I think we're good if the alert is removed. You will have to find an other way to notify the user

**Review by senner007 — CHANGES_REQUESTED** [review]

File: src/client/containers/DeleteBoardModal/DeleteBoardModal.container.js [hunk 2]
```diff
@@ -5,17 +5,25 @@ import CloseButton from '../../components/CloseButton/CloseButton.component';
...
       alert('Data deleted');
     } catch (error) {
-      throw new Error(error);
+      deleteErrorHandler(error);
```
Comment (senner007) [inline src/client/containers/DeleteBoardModal/DeleteBoardModal.container.js]: I don't see why you need to pass down a deleteHandler here.
You need to throw throw an ApiError in the catch function if its a general error that is not be handled in this component

**Review by senner007 — CHANGES_REQUESTED** [review]

File: src/client/containers/DeleteBoardModal/DeleteBoardModal.container.js [hunk 3]
```diff
@@ -22,8 +20,11 @@ const DeleteBoardModal = ({
         throw new Error(response.status);
       }
       alert('Data deleted');
-    } catch (error) {
-      deleteErrorHandler(error);
+    } catch {
+      setError(() => {
```
Comment (senner007) [inline src/client/containers/DeleteBoardModal/DeleteBoardModal.container.js:23]: You need to get the error from the catch here. 
catch (error) {

then pass the message and the status code to the ApiError

**Review by senner007 — CHANGES_REQUESTED** [review]

File: src/client/containers/DeleteBoardModal/DeleteBoardModal.container.js [hunk 4]
```diff
@@ -1,21 +1,29 @@
...
-      alert('Data deleted');
-    } catch (error) {
-      throw new Error(error);
+      if (!response.ok) {
+        throw new ApiError(response.statusText, response.status);
+      }
+    } catch (err) {
+      setError(() => {
+        // eslint-disable-next-line new-cap
+        throw new ApiError(err.message, err.statusCode);
```
Comment (senner007) [inline src/client/containers/DeleteBoardModal/DeleteBoardModal.container.js:25]: Great 👍
We really need a global fetch method to do this for us.
Try and provoke an error and check if its working. Merge when done

**Review by senner007 — APPROVED** [review]
✔

**Pull request merged**